snakemake/executors/ga4gh_tes.py

```diff
@@ -329,12 +330,12 @@ def _get_task(self, job, jobscript):
         task["resources"] = tes.models.Resources()
 
         # define resources
-        if "_cores" in job.resources:
-            task["resources"]["cpu_cores"] = job.resources["_cores"]
-        if "mem_mb" in job.resources:
-            task["resources"]["ram_gb"] = job.resources["mem_mb"] / 1000
-        if "disk_mb" in job.resources:
-            task["resources"]["disk_gb"] = job.resources["disk_mb"] / 1000
+        if job.resources.get("_cores"):
+            task.resource.cpu_cores = job.resources["_cores"]
+        if job.resources.get("mem_mb"):
+            task.resources.ram_gb = math.ceil(job.resources["mem_mb"] / 1000)
+        if job.resources.get("disk_mb"):
+            task.resources.disk_gb = math.ceil(job.resources["disk_mb"] / 1000)
```

Comment:
I think for the first case, you can really just use `.get()` in the assignment, like so:

```python
task.resources.cpu_cores = job.resource.get("_cores")
```

And for testing membership in a dictionary, the more conventional/Pythonic/readable way would be to simply write:

```python
if "mem_mb" in job.resources:
    task.resources.ram_gb = math.ceil(job.resources["mem_mb"] / 1000)
if "disk_mb" in job.resources:
    task.resources.disk_gb = math.ceil(job.resources["disk_mb"] / 1000)
```

In principle you could write these as one-liners, too, and then do the whole assignment in the model construction:

```python
task["resources"] = tes.models.Resources(
    cpu_cores = job.resource.get("_cores")
    ram_gb = math.ceil(job.resources["mem_mb"] / 1000) if "mem_mb" in job.resources else None
    disk_gb = math.ceil(job.resources["disk_mb"] / 1000) if "disk_mb" in job.resources else None
)
```

However, that is probably too unwieldy and makes it a lot harder to read. But the one-liner with `.get()` for `_cores` and the `in` operator for the RAM/disk space I think would be the cleanest.

I am wondering about another things though: Is there any particular reason you don't allow floating-point numbers for `ram_gb` and `disk_gb`? The TES specs explicitly allow them and TES clients will generally produce them. Do HPCs like Slurm, PBS etc. not support fractions of Gbs of RAM/disk resources?

Reply:
fixed